server/src/main/java/org/apache/druid/client/indexing/ClientKillUnusedSegmentsTaskQuery.java

```diff
@@ -50,16 +52,23 @@ public ClientKillUnusedSegmentsTaskQuery(
       @JsonProperty("interval") Interval interval,
       @JsonProperty("markAsUnused") @Deprecated Boolean markAsUnused,
       @JsonProperty("batchSize") Integer batchSize,
-      @JsonProperty("limit") Integer limit
+      @JsonProperty("limit") @Nullable Integer limit,
+      @JsonProperty("maxUsedFlagLastUpdatedTime") @Nullable DateTime maxUsedFlagLastUpdatedTime
   )
   {
-    this.id = Preconditions.checkNotNull(id, "id");
+    if (id == null) {
+      throw InvalidInput.exception("kill task id cannot be null");
+    }
+    if (limit != null && limit <= 0) {
```

Comment:
nit: should we add the other validation aroun batchSize and markAsUnused with limit as is being done in the task itself, or just allow the task to fail if those constraints are violated?

Reply:
Yeah, I didn't add the other validations for `batchSize` intentionally because the default batch size is being maintained by the kill task `org.apache.druid.indexing.common.task.KillUnusedSegmentsTask#DEFAULT_SEGMENT_NUKE_BATCH_SIZE`, so failing it here would break existing customers. While it'd be nice to have these validations in the client facing payload, we can move things around, I didn't want to expand the scope of the PR